crates/home/src/windows.rs

```diff
@@ -16,15 +19,19 @@ pub fn home_dir_inner() -> Option<PathBuf> {
 #[cfg(not(target_vendor = "uwp"))]
 fn home_dir_crt() -> Option<PathBuf> {
     unsafe {
-        let mut path: Vec<u16> = Vec::with_capacity(MAX_PATH as usize);
-        match SHGetFolderPathW(0, CSIDL_PROFILE as i32, 0, 0, path.as_mut_ptr()) {
+        let mut path = ptr::null_mut();
+        match SHGetKnownFolderPath(&FOLDERID_Profile, KF_FLAG_DONT_VERIFY as u32, 0, &mut path) {
             S_OK => {
-                let len = wcslen(path.as_ptr());
-                path.set_len(len);
-                let s = OsString::from_wide(&path);
+                let path_slice = slice::from_raw_parts(path, wcslen(path));
+                let s = OsString::from_wide(&path_slice);
+                CoTaskMemFree(path.cast());
```

Comment:
No, I believe that was fine because it freed the memory before the function ever returned. We supplied our own buffer for it to write into (before freeing) on success.